kotlinpoet/src/main/java/com/squareup/kotlinpoet/TypeName.kt

```diff
@@ -309,6 +309,10 @@ class ClassName internal constructor(
   constructor(packageName: String, vararg simpleNames: String) :
       this(listOf(packageName, *simpleNames)) {
     require(simpleNames.isNotEmpty()) { "simpleNames must not be empty" }
+    require(simpleNames.none { it.isEmpty() }) {
+      "simpleNames must not contain empty items: " +
+          simpleNames.joinToString(prefix = "[", postfix = "]")
```

Comment:
Kotlin's list toString() does this format automatically

Reply:
Yeah good catch, copy-pasta from the vararg constructor.